app/plugin_test.go

```diff
@@ -48,19 +48,39 @@ func TestPluginKeyValueStore(t *testing.T) {
 	assert.Nil(t, th.App.DeletePluginKey(pluginId, "postkey"))
 	assert.Nil(t, th.App.DeletePluginKey(pluginId, "notrealkey"))
 
-	// Test ListKeys
-	assert.Nil(t, th.App.SetPluginKey(pluginId, "key2", []byte("test")))
-	hashedKey := getHashedKey("key")
 	hashedKey2 := getHashedKey("key2")
+	kv := &model.PluginKeyValue{
+		PluginId: pluginId,
+		Key:      hashedKey2,
+		Value:    []byte("test"),
+		ExpireAt: 0,
+	}
+
+	result := <-th.App.Srv.Store.Plugin().SaveOrUpdate(kv)
+	assert.Nil(t, result.Err)
+
+	// Test fetch by hashed keyname
+	ret, err = th.App.GetPluginKey(pluginId, "key2")
+	assert.Nil(t, err)
+	assert.Equal(t, kv.Value, ret)
+
+	// Test fetch by keyname
+	// The key should be changed to unhashed version because of earlier GetPluginKey, so we will get nil back
```

Comment:
I don't understand this comment/test: shouldn't `hashedKey2` forever return the same result as fetching `"key2"`? Otherwise, if my plugin stores keys in an outside system, we'll lose the entry.

Reply:
HashedKey2 will be removed when we do a get for key2 because we are actively deleting hashed values to phase out old hashedKeys when a new set or get is done.